oat-sa/grunt-tao-bundle
# Feature/ADF-1417/Upgrade to Node 18

### Related to: [ADF-1417](https://oat-sa.atlassian.net/browse/ADF-1417)

### Summary

Update the dependencies in order to support further versions of Node

Also add GitHub workflow

### Details

To properly support Node 18, the dev dependencies had to be updated and aligned.

BREAKING CHANGE:
- The package now requires Node.js 14.14+
- `fs-extra` has been upgraded from `8.1.0` to `11.1.1`. The breaking change does not impact the features, it is mostly a drop of Node.js versions below `14.14`.
- `glob` is updated to version `8.1`. The `\` is no longer supported as a path separator.

**Note:** The lock file was generated with `npm@6` so it would also work seamlessly with Node 14.

It also removes vulnerabilities 🎉 
```
audited 536 packages in 2.309s

found 0 vulnerabilities
```

### How to test
- check the report on GitHub
- make sure to have Node 18 installed (you can use [nvm.sh](https://github.com/nvm-sh/nvm) for managing multiple versions)
- install the package: `npm i`
- run the unit tests: `npm test`, `npm run dev:test`
- test the linting: `npm run lint`

- from a consumer package, link it with this PR and check the behavior. Example with `tao-core`:
    - open `views/build/package.json`
    - link the PR: `"@oat-sa/grunt-tao-bundle": "https://github.com/oat-sa/grunt-tao-bundle.git#feature/ADF-1417/upgrade-to-node18",`
    - install: `npm i`
    - run the bundle task `npx grunt taobundle`

[ADF-1417]: https://oat-sa.atlassian.net/browse/ADF-1417?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
